### PR TITLE
[Admin][Committee] Replace a mandate

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/Admin/AdminCommitteeController.php [hunk 1]
```diff
@@ -94,11 +98,73 @@ public function membersAction(CommitteeManager $manager, Committee $committee):
...
+
+        $newMandateCommand = CommitteeMandateCommand::createFromCommitteeMandate($mandate);
+        $form = $this->createForm(CommitteeMandateCommandType::class, $newMandateCommand);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if ($form->get('confirm')->isClicked()) {
+                try {
+                    $mandateManager->replaceMandate($mandate, $newMandateCommand->getAdherent());
+                    $this->addFlash('sonata_flash_success', \sprintf('Le mandate avec l\'id %s a été remplacé avec succès.', $mandate->getId()));
```
Comment (ottaviano) [inline src/Controller/Admin/AdminCommitteeController.php:130]: `Le mandat`

File: src/Controller/Admin/AdminCommitteeController.php [hunk 2]
```diff
@@ -94,11 +98,73 @@ public function membersAction(CommitteeManager $manager, Committee $committee):
...
+                } catch (BaseGroupException $exception) {
+                    throw $this->createNotFoundException(\sprintf('Committee %u must not be approved in order to be approved.', $mandate->getId()), $exception);
```
Comment (ottaviano) [inline src/Controller/Admin/AdminCommitteeController.php:132]: `Committee %u` => `Committee %d` as it's for a number?

File: src/Controller/Admin/AdminCommitteeController.php [hunk 3]
```diff
@@ -94,11 +98,73 @@ public function membersAction(CommitteeManager $manager, Committee $committee):
...
+                        '%s dans le comité "%s", ',
+                        CommitteeMandateQualityEnum::SUPERVISOR === $activeMandate->getQuality()
+                            ? ($activeMandate->isProvisional() ? 'Animateur provisoire' : 'Animateur')
+                            : 'Adhérent désigné',
+                        $activeMandate->getCommittee()->getName());
+                });
+
+                $this->addFlash(
+                    'warning',
+                    substr_replace("Attention, cet adhérent est déjà $msg", '.', -2)
```
Comment (ottaviano) [inline src/Controller/Admin/AdminCommitteeController.php:152]: this warning will appear for 1 or many mandates, right?

File: src/Committee/CommitteeMandateCommand.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    public $provisional;
+
+    public static function createFromCommitteeMandate(CommitteeAdherentMandate $mandate): self
+    {
+        $dto = new self();
+        $dto->gender = $mandate->getGender();
+        $dto->committee = $mandate->getCommittee();
+        $dto->quality = $mandate->getQuality();
+        $dto->provisional = $mandate->isSupervisor() ? true : $mandate->isProvisional();
```
Comment (ottaviano) [inline src/Committee/CommitteeMandateCommand.php:51]: why not just: 
`$dto->provisional = $mandate->isSupervisor()`
or 
`$dto->provisional = (bool) $mandate->isSupervisor()`

File: src/Validator/AdherentForCommitteeMandateReplacementValidator.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        if (!$constraint instanceof AdherentForCommitteeMandateReplacement) {
+            throw new UnexpectedTypeException($constraint, CommitteeProvisionalSupervisor::class);
+        }
+
+        if (!$value) {
+            return;
+        }
+
+        if (!$value instanceof CommitteeMandateCommand) {
+            throw new UnexpectedTypeException($value, CommitteeMandateCommand::class);
```
Comment (ottaviano) [inline src/Validator/AdherentForCommitteeMandateReplacementValidator.php:32]: could you use `UnexpectedValueException` instead, please. `UnexpectedTypeException` it's for `$constraint` check

**Review by ottaviano — APPROVED** [review]

File: src/Controller/Admin/AdminCommitteeController.php (hunk 3 above)
Comment (illusionOfParadise) [reply src/Controller/Admin/AdminCommitteeController.php:152]: Yes.

**Pull request merged**